paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/complete_flow/CompleteFlowViewModel.kt

```diff
@@ -0,0 +1,110 @@
+package com.stripe.android.paymentsheet.example.samples.ui.complete_flow
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.kittinunf.fuel.Fuel
+import com.github.kittinunf.fuel.core.extensions.jsonBody
+import com.google.gson.Gson
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.paymentsheet.PaymentSheetResult
+import com.stripe.android.paymentsheet.example.samples.model.CartState
+import com.stripe.android.paymentsheet.example.samples.networking.ExampleCheckoutResponse
+import com.stripe.android.paymentsheet.example.samples.networking.toCheckoutRequest
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.flow.updateAndGet
+import kotlinx.coroutines.launch
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import kotlin.let
+import com.github.kittinunf.result.Result as ApiResult
+
+internal class CompleteFlowViewModel(
+    application: Application,
+) : AndroidViewModel(application) {
+
+    private val _state = MutableStateFlow(
+        value = CompleteFlowViewState(cartState = CartState.defaultWithHardcodedPrices),
+    )
+    val state: StateFlow<CompleteFlowViewState> = _state
+
+    fun checkout() {
+        viewModelScope.launch {
+            val currentState = _state.updateAndGet {
+                it.copy(isProcessing = true)
+            }
+
+            val request = currentState.cartState.toCheckoutRequest()
+            val requestBody = Gson().toJson(request)
+
+            val apiResult = suspendCoroutine { continuation ->
```

Comment:
Fuel has some helpers for coroutines, can we use one of those instead? (It should autocomplete if you do a `.await` at the end of the chain.